request-example.go

```diff
@@ -69,6 +75,48 @@ func (fs *root) Filewrite(r *Request) (io.WriterAt, error) {
 	return file.WriterAt()
 }
 
+func (fs *root) CheckFile(r *Request) (CheckFileResponse, error) {
+	response := CheckFileResponse{}
+	requestedAlgos := r.CheckFile.GetHashAlgoAsList()
+	var h hash.Hash
+	for _, algo := range requestedAlgos {
+		if algo == "md5" {
+			h = md5.New()
+			response.HashAlgoUsed = "md5"
+		} else if algo == "sha1" {
+			h = sha1.New()
+			response.HashAlgoUsed = "sha1"
+		} else if algo == "sha224" {
+			h = sha256.New224()
+			response.HashAlgoUsed = "sha224"
+		} else if algo == "sha256" {
+			h = sha256.New()
+			response.HashAlgoUsed = "sha256"
+		} else if algo == "sha384" {
+			h = sha512.New384()
+			response.HashAlgoUsed = "sha384"
+		} else if algo == "sha512" {
+			h = sha512.New()
+			response.HashAlgoUsed = "sha512"
+		}
+	}
+	if h == nil {
+		return response, errors.New("Unsupported hash algorithm requested")
+	}
+	// in this example we only support the full file hash
+	if r.CheckFile.StartOffset > 0 || r.CheckFile.Length > 0 || r.CheckFile.BlockSize > 0 {
```

Comment:
Nothing about `r.CheckFile.Length == 0` or `r.CheckFile.BlockSize == 0` says to me that it is a full-file hash. In such a case, you want to document what the standard says and why.